File: fsharp/raindrops/Raindrops.fs
module Raindrops

let sound n =
  match n with
  | 3 -> "Pling"
  | 5 -> "Plang"
  | 7 -> "Plong"
  | _ -> ""

let getSounds number =
  [
    for n in 1 .. number do
      if number % n = 0 then
        yield sound n 
  ]
  |> List.distinct
  |> String.concat ""

let convert (number: int): string =
  let sounds = getSounds number
  match sounds.Length with
  | 0 -> number.ToString()
  | _ -> sounds

File: csharp/raindrops/Raindrops.cs
using System;
using System.Linq;
using System.Collections;
using System.Collections.Generic;

public static class Raindrops
{
    public static string Convert(int number)
    {
        var sounds = Raindrops.GetSounds(number).ToArray();

        if (sounds.Count() == 0)
            return number.ToString();

        return string.Join(string.Empty, sounds);
    }

    private static IEnumerable<string> GetSounds(int number)
    {
        if (number % 3 == 0)
            yield return "Pling";

        if (number % 5 == 0)
            yield return "Plang";

        if (number % 7 == 0)
            yield return "Plong";

        yield break;
    }
}